### ISSUE TITLE
Firefox Lite API key security issue 

### ISSUE THREAD

**Issue by Daisy-pliu** [mozilla-tw/FirefoxLite#5251]
**Describe the bug**
the current API key has security issue as it did not come with fingerprint.

**Proposed Solution**
After discussed with dev, seems we have to create new API key with SHA certificate fingerprint to replace the current one.
(Refer https://support.google.com/firebase/answer/9137403?hl=en)

**Possible risks**
1. After create new api key we should make sure current user can still access FxLite app. 
2. After create new api key, we should make sure all Firebase settings, history, report are all preserved 
3. After create new api key, we should make sure all user can access verticals given verticals provided by backend server use API key to identify and connect FxLIite app.
4. The worst case is that once we change to a new API key, user can not use Fxlite on their phone  (which means the old key is automatically deactivated) , then we should consider force user to update to the latest version (which contians new api key).

**Comment by Daisy-pliu** [comment]
@cnevinc  could you help list possible risks that it might occur after API key change then we could email to Google Support team and consult on the solutions, thanks !

**Comment by cnevinc** [comment]
Planed test steps:
- Create a new Firebase project
and 

### Test what can the leaked key do any damage. If no, we may stop the investigation.

### Test Adding API Restriction to the Android client key
- SET the API key restriction with fingerprint
- TEST the app see if it crash
- ADD API restriction to Android key  
- TEST if the app will crash

### Test Creating A New Key and revoke the old the API
- This may not work because the server will generate a new key automatically.

**Comment by cnevinc** [comment]
> Planed test steps:
> 
> * Create a new Firebase project
>   and
> 
> ### Test what can the leaked key do any damage. If no, we may stop the investigation.
It seems we can't do anything we this Android key. Tested: Youtube, Firestore, Firebase Remote Config, Firebase Auth, Firebase Cloud Messaging API. I got below error

```
{
  "error": {
    "code": 403,
    "message": "Requests from this Android client application <empty> are blocked.",
.....
```
> ### Test Adding API Restriction to the Android client key
> * SET the API key restriction with fingerprint

As long as a fingerprint is enabled, only builds that signed with our keystore is able to get remote config and FCM messages"


> * TEST the app see if it crash

App is not crash in the above scenarios.

> * ADD API restriction to Android key
> * TEST if the app will crash
> 

Not tested

> ### Test Creating A New Key and revoke the old the API
> * This may not work because the server will generate a new key automatically.

Not tested

**Comment by cnevinc** [comment]
Some other info could be found here
https://developers.google.com/maps/api-key-best-practices
https://cloud.google.com/docs/authentication/api-keys
https://javebratt.com/hide-firebase-api/
https://medium.com/@devesu/how-to-secure-your-firebase-project-even-when-your-api-key-is-publicly-available-a462a2a58843

**Comment by cnevinc** [comment]
For Android keys, we are fine.
For other keys, we should add more restrictions.
